OSD/Imgui: Add a better default font

TryTwo opened the pull request:
Relies on: #13897 
Previous discussions: #13695

Adds VeraMono as "OSD_Font.ttf".  I was thinking we could comment somewhere that users can replace that file to use a different font if they wish.

Old 13p / New 13p / New 18p
<img width="250 " alt="image" src="https://github.com/user-attachments/assets/90988c3c-f950-4a00-a3d9-fe2eadac6b99" /> <img width="250" alt="image" src="https://github.com/user-attachments/assets/b584134f-5e12-48a0-84af-b626dc62da09" /> <img width="250" alt="image" src="https://github.com/user-attachments/assets/a5708ec5-bfd1-4267-865a-307b29d72505" />

mbc07 commented:
The main blocker for OSD translations is the default ImGui font not supporting all alphabets me might need, and it's being replaced on this PR. Wouldn't be better to jump directly to a font that has all the characters we'll need (while still supporting better sizing) instead of Bitstream Vera Sans Mono?

TryTwo commented:
In @SternXD 's Dolphin UWP work, he uses two different fonts Roboto and  NotoSans-JP (but that's 5.4mb).  I figured we'd worry about extra font stuff when a project like that is PR'd.

If someone has a font suggestion, I'd be happy to look into it, but don't know much about non-English fonts and what exactly we'd need. I feel a monospaced font - at least for numbers - is important so we don't have to over-format to deal with different character widths as numbers change.

mbc07 commented:
What about Noto Sans Mono? It's monospaced, has a free license and AFAICT has support for all languages we currently support in Dolphin...

SternXD commented:
> What about Noto Sans Mono? It's monospaced, has a free license and AFAICT has glyphs for all languages we currently support in Dolphin...

To me this sounds like the best option personally

mbc07 commented:
About making the OSD font user replaceable: it resides on Sys folder, which is normally read-only on Android and Linux (especially on Flatpak and AppImage). Could you make Dolphin look for a font override on the User folder instead? 

For example, if `User\Load\OSDFont.ttf` exists, load that, otherwise fall back to loading `Sys\Resources\OSDFont.ttf`...

TryTwo commented:
> What about Noto Sans Mono? It's monospaced, has a free license and AFAICT has glyphs for all languages we currently support in Dolphin...

You have to include the 3 Asian language fonts separately. They each seem to be about 5mb compressed.  Unfortunately, it also doesn't look as good at small ~13p font sizes.

Using system Asian fonts seems to be the obvious choice, it's just difficult.

I can add a check in the User folder.  I'm thinking this PR isn't going to solve all the font issues, but just be a step up.

mbc07 commented:
Ok, adding a new default font (even if we stick to Bitstream Vera Sans Mono for now) and handling the User folder override seems like a good initial step for me, and allow power users to directly replace the OSD font file with whatever they want. This PR LGTM once the User folder override is handled...

TryTwo commented:
Ok. Not sure what, if any, help text to provide for loading a custom font. Since it only supports English right now, and not sure if there's a convention for pointing out the User/Load location.

It is definitely useful for testing other fonts though. Can change the location if there is a better one.

mbc07 reviewed (approved):
LGTM. I think we can leave the ability to supply a custom font file omitted from the UI for now and revisit that later when we move forward with OSD translations or other ImGui-related stuff, so the current tooltip is fine for me...

JosJuice commented on Source/Core/VideoCommon/OnScreenUI.cpp:
```diff
@@ -73,11 +75,28 @@ bool OnScreenUI::Initialize(u32 width, u32 height, float scale)
-  ImGui::GetIO().BackendFlags |=
-      ImGuiBackendFlags_RendererHasTextures | ImGuiBackendFlags_RendererHasVtxOffset;
+  // User supplied font
+  std::string file = File::GetUserPath(D_LOAD_IDX) + "OSD_Font.ttf";
```
It will be non-obvious to users that they can override the VeraMono.ttf file by providing a file in the User folder with a different name. What do you think about giving it the name OSD_Font.ttf in the Sys folder too?

JosJuice replied:
>  I could add a description to OSD Font Size, to mention loading a custom font.

That's probably a good idea.

TryTwo replied:
```
<br><br>The font can be changed by placing a TTF font file into Dolphin's "
"User/Load folder, and renaming it OSD_Font.ttf." 
```
I didn't rename the file in System, let me know if you still want me to do that.

TryTwo commented:
Someone else also mentioned VeraMono.ttf should be OSD_Font.ttf in the system folder. Renamed it, and renamed the copyright file to OSD_Font_VeraMono_Copyright.

TryTwo commented:
Rebased and fixed conflicts.
